src/amplify-migration/amplifyMigrationScript.ts

```diff
@@ -344,51 +362,84 @@ export async function changeFileContent({
       fileContent = fileContent.replace(match, newMatch);
     }
 
-  
-    const { fileContent: filepathContent } =
-      await updateFilesUploadsPath(
-        url,
-        setOfAllDocumentsPath,
-        currentRepoName
-      );
+    const { fileContent: filepathContent } = await updateFilesUploadsPath(
+      url,
+      setOfAllDocumentsPath,
+      currentRepoName
+    );
     fileContent = fileContent.replace(url, filepathContent);
   }
 
-  return { fileContent, };
+  const yamlParser = YAML.parseAllDocuments(fileContent);
+
+  /**
+   * This is to handle the case where the yml file has multiple documents which are
+   * separated by document end marker lines. Since we don't expect to have > 1 yml
+   * document in a single file, we will only process the first document. The other
+   * documents in this array are expected to be null.
+   */
+  const yamlDocument = yamlParser[0];
```

Comment:
Could you add a comment about why we need to parseAllDocuments and take the first item (this is because of the front matter always being the first item when parsed)

Reply:
6cac9cd06b23ce126a03f648fe7d3c9d29af1791